test/unit/spec/DeviceFingerprint_spec.js

```diff
@@ -0,0 +1,73 @@
+/*global JSON */
+define([
+  'jquery',
+  'vendor/lib/q',
+  'helpers/util/Expect',
+  'sandbox',
+  'util/DeviceFingerprint'
+],
+function ($, Q, Expect, $sandbox, DeviceFingerprint) {
+
+  Expect.describe('DeviceFingerprint', function () {
+
+    function mockIFrameMessages(success, errorMessage) {
+      var message = success ? {
+        type: 'FingerprintAvailable',
+        fingerprint: 'thisIsTheFingerprint'
+      } : errorMessage;
+      window.postMessage(JSON.stringify(message), '*');
+    }
+
+    beforeEach(function () {
+      $sandbox.append('<div class="auth-content"></div>');
+    });
+
+    it('iframe is created with the right src and it is hidden', function () {
+      spyOn(window, 'addEventListener');
+      DeviceFingerprint.generateDeviceFingerprint('baseUrl');
+      var $iFrame = $('#okta_fingerprint_iframe');
+      expect($iFrame).toExist();
+      expect($iFrame.attr('src')).toBe('baseUrl/auth/services/devicefingerprint');
```

Comment:
I don't see any errors in the browser console either with this test... maybe because I am mocking the url base to be ```baseUrl``` and not a proper url with http protocol? 
I think in any case that the purpose of the test is still valid...